include/admin-switcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+
+class PLL_Admin_Switcher extends PLL_Switcher {
+
+	/**
+	 * Get the language elements for use in a walker
+	 *
+	 * @since 1.2
+	 *
+	 * @param PLL_Frontend_Links $links Instance of PLL_Frontend_Links.
+	 * @param array              $args  Arguments passed to {@see PLL_Switcher::the_languages()}.
+	 * @return array Language switcher elements.
+	 */
+	protected function get_elements( $links, $args ) {
+		$first = true;
+		$out   = array();
+
+		foreach ( $links->model->get_languages_list( array( 'hide_empty' => $args['hide_if_empty'] ) ) as $language ) {
+			list( $id, $order, $slug, $locale, $classes, $url ) = $this->init_foreach_language( $language );
+
+			$curlang = $links->options['default_lang'];
```

Comment:
As this is one of the only lines who changes between `PLL_Admin_Switcher` and `PLL_Frontend_Switcher`, this would need to be extracted into two methods with the same names in order to reuse the `get_elements()` method.